admin/src/scenes/centersV2/components/sessions/SessionList.jsx

```diff
@@ -0,0 +1,293 @@
+import React, { useEffect, useMemo, useState } from "react";
+import queryString from "query-string";
+import { useHistory } from "react-router-dom";
+import { useQuery } from "@tanstack/react-query";
+
+import { canPutSpecificDateOnSessionPhase1, isSessionEditionOpen } from "snu-lib";
+import { Select } from "@snu/ds/admin";
+
+import { capture } from "@/sentry";
+import { canCreateOrUpdateCohesionCenter, translate } from "@/utils";
+import api from "@/services/api";
+import dayjs from "@/utils/dayjs.utils";
+import Pencil from "@/assets/icons/Pencil";
+import ToggleDate from "@/components/ui/forms/dateForm/ToggleDate";
+import { NewGetCohortSelectOptions } from "@/services/cohort.service";
+
+import { Title } from "../commons";
+import Field from "../Field";
+import TimeSchedule from "../TimeSchedule";
+import PedagoProject from "../PedagoProject";
+import { toastr } from "react-redux-toastr";
+import OccupationCard from "./OccupationCard";
+import Loader from "@/components/Loader";
+
+function getQuerySessionId() {
+  const { sessionId } = queryString.parse(location.search);
+  if (sessionId) {
+    return sessionId;
+  } else {
+    return null;
+  }
+}
+
+export default function SessionList({ center, sessions, user, focusedSession, onFocusedSessionChange, onSessionsChange, onRefreshCenter, onCenterChange }) {
+  const history = useHistory();
+
+  const [editingBottom, setEditingBottom] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [editInfoSession, setEditInfoSession] = useState({});
+  const [errors, setErrors] = useState({});
+  const [modalDelete, setModalDelete] = useState({ isOpen: false });
+
+  const cohortList = useMemo(() => {
+    const cohorts = sessions?.map((session) => ({ name: session.cohort })) || [];
+    return NewGetCohortSelectOptions(cohorts);
+  }, [sessions]);
+
```

Comment:
pourquoi utiliser un useMemo ? on n'a pas de gros calcul a faire ca me semble overkill. 
keep it simple
  const cohortList = useMemo(() => {
    const cohorts = sessions?.map((session) => ({ name: session.cohort })) || [];
    return NewGetCohortSelectOptions(cohorts);
  }, [sessions]);

Reply:
j'ai changé la logique pour récupérer les infos des cohorts depuis redux et ensuite filtrer en fonction des sessions disponible (pour avoir les dates de sejours dans le select).
j'ai par contre gardé le memo ici car on fait une double itération (session + cohorts) et que le composant est souvent rerender (mode edition). Le memo n'aurai plus de sens si on découpait le formulaire dans un nouveau composant par exemple.